dockerpty/pty.py

```diff
@@ -249,3 +249,103 @@ def _hijack_tty(self, pumps):
                 except SSLError as e:
                     if 'The operation did not complete' not in e.strerror:
                         raise e
+
+
+class ExecPTY(PseudoTerminal):
+    def __init__(self, client, container, interactive=True, stdout=None, stderr=None, stdin=None):
+        """
+        Initialize the PTY using the docker.Client instance and container dict.
+        """
+        super(ExecPTY, self).__init__(client, container, interactive=interactive,
+                                      stdout=stdout, stderr=stderr, stdin=stdin)
+
+        self.exec_id = None
+
+
+    def exec_info(self):
+        """
+        Thin wrapper around client.inspect_container().
+        """
+        return self.client.exec_inspect(self.exec_id)
+
+
+    def exec_create(self, command):
+        self.exec_id = self.client.exec_create(self.container, command, tty=self.interactive, stdin=self.interactive)
+        return self.exec_id
```

Comment:
I wasn't sure if it should be stateless or not so I made it as generic as possible. Will rewrite then.